fieldmaptrack/track.py

```diff
@@ -94,23 +96,33 @@ def calc_reference_point(self):
         b = 0.5*((x1+x2) - a * (z1+z2))
         return (b, 0.0)
 
+    def calc_field(self, rx, ry, rz):
+        """."""
+        if self.radia_model is not None:
+            bfield = self.radia_model.get_field_at_point([rx, ry, rz])
+            bx, by, bz = bfield
+            # print("Getting field at point: {}, {}, {}".format(rx, ry, rz))
+        else:
```

Comment:
In order to decrease identation level, consider doing this:
```python3
        if self.radia_model is not None:
            return self.radia_model.get_field_at_point([rx, ry, rz])
        
        try:
            return self.fieldmap.interpolate(rx, ry, rz)
        except (fieldmap.OutOfRangeRangeRx, 
                fieldmap.OutOfRangeRxMin,
                fieldmap.OutOfRangeRy):
            rstr = 'extrapolation at ' + str((rx, ry, rz))
            if self.not_raise_range_exceptions:
                print(rstr)
                return 0.0, 0.0, 0.0
            raise TrackException(rstr)
        except fieldmap.OutOfRangeRz:
            return 0.0, 0.0, 0.0
```

this approach makes the code shorter, cleaner and easy to read.